### PR TITLE
handle grant authz for restricted marker

### PR DESCRIPTION
**Pull request by leeduan**
https://github.com/provenance-io/digital-currency-consortium/issues/53

### CODE REVIEW

File: service/src/main/kotlin/io/provenance/digitalcurrency/consortium/service/PbcService.kt [hunk 1]
```diff
@@ -203,4 +211,28 @@ class PbcService(
...
+            signers = listOf(BaseReqSigner(managerKey)),
+            txBody = MsgGrant.newBuilder()
+                .setGranter(managerAddress)
+                .setGrantee(provenanceProperties.contractAddress)
+                .setGrant(
+                    Grant.newBuilder()
+                        .setExpiration((expiration ?: maxExpiration).toProtoTimestamp())
+                        .setAuthorization(
+                            MarkerTransferAuthorization.newBuilder()
+                                .also { builder -> coins.forEach { builder.addTransferLimit(it) } }
```
Comment (scirner22) [inline service/src/main/kotlin/io/provenance/digitalcurrency/consortium/service/PbcService.kt:228]: is there a `addAllTransferLimit` function?

Comment (leeduan) [reply service/src/main/kotlin/io/provenance/digitalcurrency/consortium/service/PbcService.kt:228]: doh there has to be

**Review by scirner22 — APPROVED** [review]

**Review by ranadewall — APPROVED** [review]

File: service/src/main/kotlin/io/provenance/digitalcurrency/consortium/service/PbcService.kt [hunk 2]
```diff
@@ -203,4 +211,28 @@ class PbcService(
                 .toAny()
                 .toTxBody()
         ).throwIfFailed("Accept failed")
+
+    private val maxExpiration = OffsetDateTime.of(2031, 1, 1, 0, 0, 0, 0, ZoneOffset.UTC)
```
Comment (jtalis) [inline service/src/main/kotlin/io/provenance/digitalcurrency/consortium/service/PbcService.kt:215]: Should this be relative? Maybe just plus years from current date

**Review by jtalis — APPROVED** [review]

File: service/src/main/kotlin/io/provenance/digitalcurrency/consortium/service/PbcService.kt (hunk 2 above)
Comment (leeduan) [reply service/src/main/kotlin/io/provenance/digitalcurrency/consortium/service/PbcService.kt:215]: yeah probably better to just do now() + 10 years

Comment (jtalis) [reply service/src/main/kotlin/io/provenance/digitalcurrency/consortium/service/PbcService.kt:215]: long live the consortium

**Pull request merged**